https://github.com/open-mpi/ompi/commit/927e9aa97373dac652f9cba4813e6ee609ca2830
https://bugs.gentoo.org/828123

From: George Bosilca <bosilca@icl.utk.edu>
Date: Tue, 23 Nov 2021 14:28:11 -0500
Subject: [PATCH] Fix compile failure with enable-heterogeneous.

An internal header file was missing, leading to an incomplete type.

Fixes #9697.

Signed-off-by: George Bosilca <bosilca@icl.utk.edu>
--- a/ompi/mca/coll/base/coll_base_alltoall.c
+++ b/ompi/mca/coll/base/coll_base_alltoall.c
@@ -29,6 +29,7 @@
 #include "mpi.h"
 #include "ompi/constants.h"
 #include "ompi/datatype/ompi_datatype.h"
+#include "opal/datatype/opal_convertor_internal.h"
 #include "ompi/communicator/communicator.h"
 #include "ompi/mca/coll/coll.h"
 #include "ompi/mca/coll/base/coll_tags.h"
--- a/ompi/mca/coll/base/coll_base_alltoallv.c
+++ b/ompi/mca/coll/base/coll_base_alltoallv.c
@@ -31,6 +31,7 @@
 #include "mpi.h"
 #include "ompi/constants.h"
 #include "ompi/datatype/ompi_datatype.h"
+#include "opal/datatype/opal_convertor_internal.h"
 #include "ompi/communicator/communicator.h"
 #include "ompi/mca/coll/coll.h"
 #include "ompi/mca/coll/base/coll_tags.h"
